### PR TITLE
issue_#132 Show game number in schedule view

### PR DESCRIPTION
**Pull request by ViznyakS**
### Summary

Each game has a number. Display  number along with game on Schedule page, on the left side in the midel of teams name.

Modify view:
- _PlayoffGameResult
- _ScheduleGameResult

### Areas Of Interest

Should I create partial view to display number in PlayOff, because thare is a duplicated code.

#### Design

- [x] Web API layer contains as little logic as possible
- [x] Domain objects use semantic names

#### Perfromance

- [x] In case of any DB query was changed: attach generated SQL for review
- [x] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [x] Naming: Initial state and expected result are properly stated
- [x] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [x] Test is Isolated
- [x] Mock instances and expected instances do not share references

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 1]
```diff
@@ -112,23 +145,36 @@
...
+                                </div>
+                                <div class="col-md-2">
+                                    <strong>  @Model.GameScore.Away </strong>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
                 </div>
             </div>
+            <p>here</p>
```
Comment (MikolaB) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: What is 'here' for?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 2]
```diff
@@ -147,21 +193,33 @@
...
+                    </div>
+                    <div class="col-md-7">
+                        <div class="row">
+                            <div class="row">
+                                <div class="col-md-10">
+                                    <strong>  @Model.HomeTeamName </strong>
+                                </div>
+                                <div class="col-md-2">
+                                    <strong>   @Model.GameScore.Home </strong>
+                                </div><
```
Comment (MikolaB) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Please remove '<'.

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 3]
```diff
@@ -4,15 +4,23 @@
...
-                <strong>  @Resources.UI.TournamentViews.HomeTeamPlaceholder </strong>
-            </div>
-        </div>
-
-        <div class="row">
-            <div class="col-md-10">
-                <strong>  @Resources.UI.TournamentViews.AwayTeamPlaceholder </strong>
+        <div class="container-fluid">
+            <div class="row">
+                <div class="col-md-3" style="margin-top:8px">
```
Comment (MikolaB) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Move inline styles into css file.

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 1 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: ups

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 2 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: ok

**Comment by sdiachen** [comment]
Please undo any changes to checklist section so I can properly review changes

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 4]
```diff
@@ -1,18 +1,28 @@
 ﻿@model VolleyManagement.UI.Areas.Mvc.ViewModels.GameResults.GameResultViewModel
 
+<link rel="stylesheet" href="~/Content/Site.css" />
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: What is this for?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
-        </div>
-
-        <div class="row">
-            <div class="col-md-10">
-                <strong>  @Resources.UI.TournamentViews.AwayTeamPlaceholder </strong>
+        <div class="container-fluid">
+            <div class="row">
+                <div class="col-md-3 game-number">
+                    <div>
+                        <strong>@Html.Raw("G" + @Model.GameNumber) </strong>
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Such formatting is a responsibility of view model

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 4 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: To connect styles

Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Weren't they already working?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 5 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Should I change int to string and add "G" in property( public int GameNumber { get; set; }) or add new property FormatGameNumber(for example)?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 4 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: No, not work.

Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Hm... Indeed it is not working. Remove it from here and add Site.css to the "commoncss" bundle.

Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: "commoncss" bundle???

Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: https://github.com/VolleyManagement/volley-management/blob/master/src/VolleyManagement.Backend/VolleyManagement.UI/App_Start/BundleConfig.cs

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 6]
```diff
@@ -81,6 +81,21 @@ public GameResultViewModel()
...
+        /// <summary>
+        /// Gets or sets the number of the game in the tournament
+        /// </summary>
+        public string FormatedGameNumber
+        {
+            get
+            {
+                return "G" + ((int)GameNumber).ToString();
+            }
+            set
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: BLOCKER Use the 'value' parameter in this property set accessor declaration. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3237)

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 7]
```diff
@@ -81,6 +81,21 @@ public GameResultViewModel()
...
+            {
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: MAJOR Either remove or fill this block of code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS108)

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml (hunk 4 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: Thanks)

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 8]
```diff
@@ -81,6 +81,17 @@ public GameResultViewModel()
...
         public byte GameNumber { get; set; }
 
+        /// <summary>
+        /// Gets formated number of the game in the tournament
+        /// </summary>
+        public string FormatedGameNumber
+        {
+            get
+            {
+                return "G" + ((int)GameNumber).ToString();
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: 1. Why cast to int?

2. Where are UTs to check this logic?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 9]
```diff
@@ -81,6 +81,17 @@ public GameResultViewModel()
         /// </summary>
         public byte GameNumber { get; set; }
 
+        /// <summary>
+        /// Gets formated number of the game in the tournament
+        /// </summary>
+        public string FormatedGameNumber
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: "DisplayGameNumber"

File: src/VolleyManagement.Backend/VolleyManagement.UI/App_Start/BundleConfig.cs [hunk 10]
```diff
@@ -31,7 +31,9 @@ private static void RegisterCommonBundles(BundleCollection bundles)
         {
             bundles.Add(new StyleBundle("~/bundles/commoncss")
                 .Include(
-                    "~/Content/bootstrap.min.css"));
+                    "~/Content/bootstrap.min.css")
+                .Include(
+                    "~/Content/Site.css"));
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.UI/App_Start/BundleConfig.cs]: MINOR Refactor your code not to use hardcoded absolute paths or URIs. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1075)

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 11]
```diff
@@ -81,6 +81,17 @@ public GameResultViewModel()
         /// </summary>
         public byte GameNumber { get; set; }
 
+        /// <summary>
+        /// /// Gets display number of the game in the tournament
```
Comment (dmshapoval) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: double slashes

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml [hunk 12]
```diff
@@ -4,15 +4,23 @@
...
-            </div>
-        </div>
-
-        <div class="row">
-            <div class="col-md-10">
-                <strong>  @Resources.UI.TournamentViews.AwayTeamPlaceholder </strong>
+        <div class="container-fluid">
+            <div class="row">
+                <div class="col-md-3 game-number">
+                    <div>
```
Comment (dmshapoval) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Views/Tournaments/_PlayoffGameResult.cshtml]: what the reason of putting inside empty div here and in all other places? if you need some specific styling (like display:block or display: inline-block solve this with styling not markup)

File: src/VolleyManagement.Backend/VolleyManagement.UI/App_Start/BundleConfig.cs (hunk 10 above)
Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.UI/App_Start/BundleConfig.cs]: You can ignore this error

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs [hunk 13]
```diff
@@ -81,6 +81,17 @@ public GameResultViewModel()
         /// </summary>
         public byte GameNumber { get; set; }
 
+        /// <summary>
+        /// Gets display number of the game in the tournament
+        /// </summary>
+        public string DisplayGameNumber
```
Comment (MikolaB) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: You can use Expression-Bodied Properties

File: src/VolleyManagement.Backend/VolleyManagement.UI/Content/Site.css [hunk 14]
```diff
@@ -149,6 +149,10 @@ div.game-away-team {
     text-align: left;
 }
 
+div.game-number {
+    margin-top:8px;
+}
```
Comment (MikolaB) [inline src/VolleyManagement.Backend/VolleyManagement.UI/Content/Site.css]: Please format.

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs [hunk 15]
```diff
@@ -91,9 +92,12 @@ public void CreatePostAction_ValidModel_Created()
 
             // Act
             var result = sut.Create(gameResult) as RedirectToRouteResult;
+            string expectedDisplayGameNumber = "G" + gameResult.GameNumber;
 
             // Assert
             TestHelper.AreEqual(expectedGameResult, actualGameResult, new GameComparer());
+            Assert.AreEqual(expectedDisplayGameNumber, gameResult.DisplayGameNumber);
```
Comment (sdiachen) [inline tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Why existing code did not suit you?

File: src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs (hunk 13 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/GameResults/GameResultViewModel.cs]: ok

File: src/VolleyManagement.Backend/VolleyManagement.UI/Content/Site.css (hunk 14 above)
Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.UI/Content/Site.css]: ok

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs (hunk 15 above)
Comment (ViznyakS) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: What do you mean? I add new 3 line to check if the DisplayGameNumber return correct data and to check if DisplayGameNumbe not equal GameNumber.

Comment (ViznyakS) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Existing code don't check if property DisplayGameNumber return correct data, so that's why I add 3 new line to check it.  Of courese I can add in GameComparer to check if the DispalyGameNumber of one games equal to another, but when if there are mistake of logic of displaying number of game, the test don't fail.

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs [hunk 16]
```diff
@@ -91,9 +92,10 @@ public void CreatePostAction_ValidModel_Created()
 
             // Act
             var result = sut.Create(gameResult) as RedirectToRouteResult;
-
+          
             // Assert
             TestHelper.AreEqual(expectedGameResult, actualGameResult, new GameComparer());
+
```
Comment (sdiachen) [inline tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: ?

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs [hunk 17]
```diff
@@ -134,6 +136,8 @@ public void CreatePostAction_InvalidModel_ExceptionThrown()
                                 .WithAwayTeamId(HOME_TEAM_ID)
                                 .Build();
 
+            gameResult.GameNumber = 5;
```
Comment (sdiachen) [inline tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Why did you add it here?

Comment (ViznyakS) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: In Builder I initialise GameNumber by 0, if I change in Builder to another number, two another test faild, because someone expected data with GameNumber equal 0(GameResultDtoBuilder initialize GameNumber by 0)
To make test more difficult) Remove it?

Comment (sdiachen) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: This test verifies invalid model scenario - how game number affects it?

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs (hunk 16 above)
Comment (sdiachen) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Why empty line?

File: tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs (hunk 17 above)
Comment (ViznyakS) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Each game has a number, when user create invalid model, the error message say what to change, but we return current model to user to modify.

Comment (ViznyakS) [reply tests/VolleyManagement.UnitTests/Mvc/Controllers/GameResultsControllerTests.cs]: Can I create new test?
